src/js/components/CheckBoxGroup/CheckBoxGroup.js

```diff
@@ -103,7 +104,9 @@ export const CheckBoxGroup = forwardRef(
               onChange={(event) =>
                 onCheckBoxChange(event, valueOption, optionProps)
               }
-            />
+            >
+              {children ? (state) => children(option.value, state) : null}
+            </CheckBox>
```

Comment:
This doesn't work when options is an array of objects: `options={[{label: 'asc', disabled: true},{label: 'desc'}]}`

Reply:
I can solve this by passing `options[index]` into the children.
One more thing for the array of objects that you have shown in above: `options={[{label: 'asc', disabled: true},{label: 'desc'}]}`.When I put this in options its showing me a bug by checking both the checkboxes when I select one. Correct me if I am wrong I think this is because we didn't pass `value` prop along with label. I am not sure if the checkboxes are supposed perfectly when only `label` prop with array of objects is passed. Below is the screenshot of above `options` object.

![1](https://user-images.githubusercontent.com/61521805/126592299-5c17587a-262e-4be6-afb2-6cf5d6d1987e.PNG)
![2](https://user-images.githubusercontent.com/61521805/126592301-5110919a-8040-4ed6-9f86-80e3a8ced86e.PNG)

Reply:
I think it is fine to have the object in the options array not include a 'value' prop but then you need to include a 'valueLabel' prop in CheckBoxGroup. See the [CheckBoxGroup/Form Uncontrolled](https://storybook.grommet.io/?path=/story/input-checkboxgroup-form-uncontrolled--form-uncontrolled) and [CheckBoxGroup/Object](https://storybook.grommet.io/?path=/story/input-checkboxgroup-object-options--object-options) Options stories for examples

Reply:
okay, Thanks